File: atlas-operations-command/backend/app/schemas/factory.py
from pydantic import BaseModel, ConfigDict, Field
from typing import Optional, List
from datetime import datetime, date
from uuid import UUID
from decimal import Decimal

# -----------------
# Factory Schemas
# -----------------
class FactoryBase(BaseModel):
    name: str = Field(..., max_length=255)
    code: Optional[str] = Field(None, max_length=100)
    location: Optional[str] = Field(None, max_length=255)
    status: str = Field("active", max_length=50)

class FactoryCreate(FactoryBase):
    pass

class FactoryResponse(FactoryBase):
    id: UUID
    company_id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class FactoryFinancialsResponse(BaseModel):
    factory_id: UUID
    factory_name: str
    month: int
    year: int
    revenue: Decimal
    expenditure: Decimal
    profit: Decimal

File: atlas-operations-command/Frontend/src/types/factory.ts
/**
 * Factory Domain Types
 * Exactly mirrors backend Phase 3 schemas:
 * - FactoryResponse: app/schemas/factory.py (FactoryResponse)
 * - FactoryFinancialsResponse: app/schemas/factory.py (FactoryFinancialsResponse)
 */

export interface Factory {
  id: string;
  name: string;
  code?: string | null;
  location?: string | null;
  status: 'active' | 'maintenance' | 'inactive' | string;
  created_at?: string;
  updated_at?: string;
}

export interface FactoryFinancials {
  factory_id: string;
  factory_name: string;
  month: number;
  year: number;
  revenue: number;
  expenditure: number;
  profit: number;
}
